src/main/java/echoserver/ClientServerConnection.java

```diff
@@ -0,0 +1,25 @@
+package echoserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.Socket;
+
+public class ClientServerConnection implements Connection{
+    public final Socket socket;
```

Comment:
Does this need to be public?

Reply:
Probably not and will also close the socket connection.

Reply:
I think we need as public since we're checking if the socket is open

Reply:
Is there a way to check if it is open without widening the public interface?

Reply:
we could have a method on ClientServerConnection but that in essence widens the interface.  A separate method or making socket public - both seem to widen the interface?

Reply:
You may have tried this already, but I might look to see if you can override `close` on your `FakeSocket`.